test/functional/page_objects/solution_navigation.ts

```diff
@@ -76,7 +76,9 @@ export function SolutionNavigationProvider(ctx: Pick<FtrProviderContext, 'getSer
             timeout: TIMEOUT_CHECK,
           });
         } else {
-          expect(await getByVisibleText('~nav-item', by.text)).not.be(null);
+          expect(await (await getByVisibleText('~nav-item', by.text))?.getVisibleText()).to.be(
```

Comment:
Why did we need this change?

We should probably introduce a different function if we want to change this assertion to do an expect on the full visible text. The existing assertion is getting any nav item with the text and assert we found something. This is asserting you found a nav item with exactly the text provided. Those are similar but subtly different. I'm also confused by why we are getting an element by the text, then calling `.getVisibleText()` on it.


Additionally given what we changed it to here I would have introduced an `else if ('text' in by)` clause for this assertion specifically instead of just doing it in the  catch-all `else`.

Reply:
With the previous condition in a broken test we would get "null" not equal "null". And for now we have at least name what exactly broken "null" not equal "Overview" for example. I can rollback it if we need

Reply:
it looks like we're the only ones using `expectLinkExists({ text: '' })` BUT it is also used in a lot of the other helpers, so I'm hesitant to change the behavior. Granted if the FTRs are passing maybe we didn't break anything.

We could updated our tests to check using the `expectLinkExists({ deepLinkId: '' })` if thats more reliable than using `getByVisibleText`.

Reply:
I guess this work is better to do in complex so I revert my changes